spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindUnrelatedTypesInGenericContainer.java

```diff
@@ -846,53 +846,6 @@ private IncompatibleTypes compareTypeParameters(GenericObjectType parmGeneric, G
         return IncompatibleTypes.SEEMS_OK;
     }
 
-    // old version of compare types
-    private boolean compareTypesOld(Type parmType, Type argType) {
```

Comment:
Two bugs, you are right once again. @hazendaz

You argue with perfection, which is kind of odd, some even consider obsolete, as we never get there.

I did the same and failed miserably many times. So I beg you to stop demanding it, as it's never reached in this dimension. 

All we can do is try, layering as many solid tools there are.

@timtebeek i tag you as i know you **are** passioned about **it** just as our common friend is. He´s a strong advocate, and constructive critic, pro and contra rewrite, just as me.

2 changes to make it better, improving for the whole community.

- https://docs.openrewrite.org/recipes/staticanalysis/removeunusedprivatemethods

Hope we can work this out. As PMD is kind of only possible by using rewrite.